java/src/jmri/jmrit/display/layoutEditor/LayoutTrack.java

```
@@ -181,40 +176,166 @@ protected boolean isConnectionType(int connectionType) {
+            case TURNOUT_B:
+            case TURNOUT_C:
+            case TURNOUT_D: {
+                ((LayoutTurnout)this).setConnection(connectionType, o, type);
+                break;
+            }
+            case LEVEL_XING_A:
+            case LEVEL_XING_B:
+            case LEVEL_XING_C:
+            case LEVEL_XING_D: {
+                ((LevelXing)this).setConnection(connectionType, o, type);
+                break;
+            }
+
+            case TRACK: {
+                ((TrackSegment)this).setConnection(connectionType, o, type);
+                break;
+            }
+
+            case SLIP_A:
+            case SLIP_B:
+            case SLIP_C:
+            case SLIP_D: {
+                ((LayoutSlip)this).setConnection(connectionType, o, type);
+                break;
+            }
+            default: {
+                if (connectionType >= TURNTABLE_RAY_OFFSET) {
+                    ((LayoutTurntable)this).setConnection(connectionType, o, type);
+                } else {
+                    log.error("Invalid connection type " + connectionType); //I18IN
+                    throw new jmri.JmriException("Invalid Point");
+                }
+                break;
+            }
+        }
     }
 
     /**
-     * @return the bounds of this track
+     * return true if this connection type is disconnected
+     * @param connectionType - the connection type to test
+     * @return true if the connection for this connection type is free
      */
-    public Rectangle2D getBounds() {
-        log.error("virtual method: override in sub-classes and don't call super...].");
-        return new Rectangle2D.Double();
+    public boolean isDisconnected(int connectionType) {
+        boolean result = false;
+        if (isConnectionType(connectionType)) {
+            try {
+                result = (null == getConnection(connectionType));
+            } catch (Exception e) {
+                //exceptions make me throw up...
```

I would really prefer a comment explaining why the Exception can be ignored, or actually logging the exception (since everything that can throw an Exception is providing a descriptive cause in the Exception, it can be logged once here instead of in each method that could cause it). Also please only catch JmriExceptions in this case, since those are the only exceptions documented as thrown by ```getConnection()```.